src/TTImp/ProcessBuiltin.idr

```diff
@@ -0,0 +1,163 @@
+||| Checking a %builtin pragma is correct.
+-- If we get more builtins it might be wise to move each builtin to it's own file.
+module TTImp.ProcessBuiltin
+
+import Libraries.Data.Bool.Extra
+import Libraries.Data.NameMap
+import Data.List
+
+import Core.Core
+import Core.Context
+import Core.Context.Log
+import Core.Env
+import Core.Metadata
+import Core.TT
+import Core.UnifyState
+
+import TTImp.TTImp
+
+||| Get the return type.
+getRetTy : {vars : _} -> Term vars -> (vars ** Term vars)
+getRetTy (Bind _ _ _ tm) = getRetTy tm
+getRetTy tm = (vars ** tm)
```

Comment:
I suspect you're going to want to `subst` in case the `Bind` is a `Let`.

Reply:
Could you explain the `Let` constructor of `Core.TT.Binder` please?

Reply:
Actually I think I get it, is it
`Bind fc x (Let _ _ val ty) scope` represents
```
let x : ty
    x = val in
scope
```